lib/assert.js

```diff
@@ -70,19 +70,11 @@ x.notSame = function (val, expected, msg) {
 
 x.throws = function (fn, err, msg) {
 	if (fn && fn.then) {
-		var isFailed = false;
-
 		return fn
-			.catch(function (err) {
-				isFailed = true;
-
-				return err;
+			.then(function () {
+				x.throws(function () {}, err, msg);
 			})
-			.then(function (fnErr) {
-				if (!isFailed) {
-					x.throws(function () {}, err, msg);
-				}
-
+			.catch(function (fnErr) {
```

Comment:
In what situation does line 75 throw an error? It seems you are correct though, if I change it to your suggestion, it works.

```js
			.then(function () {
				x.throws(function () {}, err, msg);
			}, function (fnErr) {
				x.throws(function () {
					throw fnErr;
				}, err, msg);
			});
```